corehq/apps/userreports/expressions/factory.py

```diff
@@ -160,6 +162,57 @@ def _evaluator_expression(spec, context):
     return wrapped
 
 
+def _get_forms_expression(spec, context):
+    wrapped = FormsExpressionSpec.wrap(spec)
+    wrapped.configure(
+        case_id_expression=ExpressionFactory.from_spec(wrapped.case_id_expression)
+    )
+    return wrapped
+
+
+def _filter_items_expression(spec, context):
+    wrapped = FilterItemsExpressionSpec.wrap(spec)
+    wrapped.configure(
+        items_expression=ExpressionFactory.from_spec(wrapped.items_expression),
+        filter_expression=_make_filter(wrapped.filter_expression, context)
+    )
+    return wrapped
+
+
+def _map_items_expression(spec, context):
+    wrapped = MapItemsExpressionSpec.wrap(spec)
+    wrapped.configure(
+        items_expression=ExpressionFactory.from_spec(wrapped.items_expression),
+        map_expression=ExpressionFactory.from_spec(wrapped.map_expression)
+    )
+    return wrapped
+
+
+def _reduce_items_expression(spec, context):
+    wrapped = ReduceItemsExpressionSpec.wrap(spec)
+    wrapped.configure(
+        items_expression=ExpressionFactory.from_spec(wrapped.items_expression)
+    )
+    return wrapped
+
+
+def _flatten_expression(spec, context):
+    wrapped = FlattenExpressionSpec.wrap(spec)
+    wrapped.configure(
+        items_expression=ExpressionFactory.from_spec(wrapped.items_expression)
+    )
+    return wrapped
+
+
+def _sort_items_expression(spec, context):
```

Comment:
I played with making `ExpressionProperty` and `FilterProperty` jsonobject field types but had an issue with the context. I agree would be nice to clean this up though and also agree is out of scope.